source/agora/consensus/state/ValidatorSet.d

```diff
@@ -307,6 +307,38 @@ public class ValidatorSet
         }
     }
 
+    /***************************************************************************
+
+        Get all the current validators in ascending order of public key
+
+        Params:
+            pub_keys = will contain the public keys
+            height = the block height for which we want the active validators
+
+        Returns:
+            Return true if there was no error in getting the public keys
+
+    ***************************************************************************/
+
+    public bool getActiveValidatorPublicKeys (ref PublicKey[] pub_keys, Height height)
+        @trusted nothrow
+    {
+        try
+        {
+            assumeSafeAppend(pub_keys);
```

Comment:
Please also set the length to zero before doing this. Otherwise we have leaky abstractions because every call site needs to know to set the length to zero before calling the function.